src/m365/commands/setup.ts

```diff
@@ -106,59 +148,215 @@ class SetupCommand extends AnonymousCommand {
 
     const preferences: Preferences = {};
 
-    const usageModeConfig: SelectionConfig<string> = {
+    if (!args.options.skipApp) {
+      const entraAppConfig: SelectionConfig<EntraAppConfig> = {
+        message: 'CLI for Microsoft 365 requires a Microsoft Entra app. Do you want to create a new app registration or use an existing one?',
+        choices: [
+          { name: 'Create a new app registration', value: EntraAppConfig.Create },
+          { name: 'Use an existing app registration', value: EntraAppConfig.UseExisting },
+          { name: 'Skip configuring app registration', value: EntraAppConfig.Skip }
+        ]
+      };
+      preferences.entraApp = await cli.promptForSelection(entraAppConfig);
+      switch (preferences.entraApp) {
+        case EntraAppConfig.Create:
+          const newEntraAppScopesConfig: SelectionConfig<NewEntraAppScopes> = {
+            message: 'What scopes should the new app registration have?',
```

Comment:
I went with scopes to make it clear that it's delegated permissions rather than roles which would be app permissions. Permissions is more common but less clear to those who understand the difference